### PR TITLE
Install [docs] dependencies in setup.py

### LINKED ISSUE #3435: Documentation for Pushover reporter says "sends email"
The documentation for the [Pushover reporter](https://docs.buildbot.net/current/manual/cfg-reporters.html#pushover-notifications) talks about sending emails:

> The following simple example will send an email upon the completion of each build. The email contains a description of the Build, its results, and URLs where more information can be obtained.

It should probably talk about sending a Pushover message or something like that.

### PR DESCRIPTION
**Pull request by adityadivekar03**
Remove the pip install command from the docs Makefile used for installing the sphinx distribution.
Add the distribution to the extras in setup.py so that it can be installed if required by the user.
Add condition in the conf.py file to check if the [docs] dependencies are installed. If not, suitable error is displayed.

fixes #3435

### CODE REVIEW

**Comment by tardyp** [comment]
please look at travis results:

docs/conf.py:25:5: F401 'sphinxcontrib' imported but unused
    import sphinxcontrib.blockdiag
    ^
1       F401 'sphinxcontrib' imported but unused

**Comment by adityadivekar03** [comment]
The import statement has been put in place to check if the user has the sphinxcontrib.blockdiag package installed. If not we need to raise an error, since we require it for documentation.
The import is thus used only for checking purposes. 
Will it be an issue to have an unused import ?

File: master/docs/conf.py [hunk 1]
```diff
@@ -21,7 +21,18 @@
 sys.path.append(os.path.abspath('.'))
 
 # -- General configuration -----------------------------------------------------
-
+try:
+    import sphinxcontrib.blockdiag
```
Comment (rutsky) [inline master/docs/conf.py]: add `  # noqa` here to ignore flake8 linting errors (with two spaces before `#` sign):

```
    import sphinxcontrib.blockdiag  # noqa
```

File: .travis.yml [hunk 2]
```diff
@@ -130,7 +130,7 @@ install:
   # Install sphinx so that pylint will be able to import it
   - "[ $TESTS != lint ] || pip install sphinx==1.3.3"
   # TODO: Pin Sphinx version to workaround http://trac.buildbot.net/ticket/3408
```
Comment (rutsky) [inline .travis.yml]: Please move this comment (`TODO: Pin Sphinx ...`) to the corresponding line in `setup.py`.

File: rtd-requirements.txt [hunk 3]
```diff
@@ -4,7 +4,4 @@
 -e master/
```
Comment (rutsky) [inline rtd-requirements.txt]: Please, place `master[docs]` here, and remove whole paragraph below.

**Comment by rutsky** [comment]
One more rebase needed...

Changes LGTM with the exceptions that I commented.
Good work, @adityadivekar03!

**Comment by adityadivekar03** [comment]
Rebased and pushed :)
@rutsky  I apologize for the dragging of this issue for so long.

**Comment by rutsky** [comment]
LGTM, lets wait for CI to finish and merge.

Thanks, @adityadivekar03 !

**Comment by adityadivekar03** [comment]
The python 2.6 tests are failing it seems. I'm not able to understand the traceback. Do you have any ideas?

**Comment by rutsky** [comment]
Yes, this is https://github.com/crossbario/autobahn-python/issues/614.
Lets ignore these py 2.6 errors for now.

**Comment by oberstet** [comment]
@rutsky Yep, Autobahn does not any longer support Python 2.6 (we once did ..): https://github.com/crossbario/autobahn-python/blob/master/setup.py#L224 and https://github.com/crossbario/autobahn-python/blob/master/.travis.yml#L10

Is that a problem for BB?

**Comment by rutsky** [comment]
> Is that a problem for BB?

I don't think so.

Buildbot still runs on Python  2.6, but to do so we already use quite old version of Twisted (Twisted also dropped support for 2.6 some time ago).
We will pin maximum autobahn version on Python 2.6, as we done with Twisted, so this is not a problem.

Honestly I don't think even Python 2.6 support is feasible, but while it's not (yet) time consuming I think we will support it.

**Comment by rutsky** [comment]
Workaround for Python 2.6 (or whole Py 2.6 dropping) is discussed in #2048 .

**Comment by adityadivekar03** [comment]
Seems both pylint and flake8 are raising import errors. 

How about we use a statement `assert sphinxcontrib-blockdiag` just to show that the import is being used in some way? We can place this statement in the try block.

**Comment by rutsky** [comment]
flake8 doesn't report errors, only pylint does.

@adityadivekar03 can you try this?

```
    import sphinxcontrib.blockdiag as _  # noqa
```

I think pylint ignores unused `_` variable, e.g. here: https://github.com/buildbot/buildbot/blame/master/master/buildbot/test/__init__.py#L64

**Comment by rutsky** [comment]
Also we could try to ignore unused imports for that specific line, like as done [here](http://stackoverflow.com/a/29562085/391865), but I'm not sure how to put both pylint's `# pylint: disable=unused-import` and flake8's `noqa` on the same line.

**Comment by adityadivekar03** [comment]
I too came across the similar solution but couldn't figure out a way to include both the instructions in the same line. 
But using the statement `assert sphinxcontrib.blockdiag` solved the issue. Now it is no more an unused import. That should solve it :)

**Comment by rutsky** [comment]
Personally I would prefer to be explicit: setting `# noqa` or `# pylint: ...` is explicit declaration that you intent to ignore some warning, and `assert` should be used for assertions.

But your approach is ok, since pylint + flake8 doesn't give us much choices, so lets merge it.

Thanks, @adityadivekar03 !

**Comment by adityadivekar03** [comment]
Yes, I'll keep that practice in mind. 
@rutsky I should say thanks to you for being tolerant for so long!

File: master/docs/conf.py (hunk 1 above)
Comment (tardyp) [inline master/docs/conf.py]: @rutsky what we usually do in the code is  
```
import sphinxcontrib.blockdiag
[sphinxcontrib.blockdiag]
```

problem with assert is that it evaluates the Trueness of the thing which could actually do some side effects or even return false

**Pull request merged**